pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/defaultcolumn/BaseDefaultColumnHandler.java

```diff
@@ -583,15 +583,31 @@ private void createDerivedColumnV1Indices(String column, FunctionEvaluator funct
       Object[] outputValues = new Object[numDocs];
       PinotDataType outputValueType = null;
       for (int i = 0; i < numDocs; i++) {
+        boolean hasNull = false;
         for (int j = 0; j < numArguments; j++) {
           inputValues[j] = valueReaders.get(j).getValue(i);
+          Object defaultNullValue = argumentsMetadata.get(j).getFieldSpec().getDefaultNullValue();
+          if (inputValues[j] == null || inputValues[j] == defaultNullValue) {
+            hasNull = true;
+            break;
+          }
         }
-        Object outputValue = functionEvaluator.evaluate(inputValues);
-        outputValues[i] = outputValue;
-        if (outputValueType == null) {
-          Class<?> outputValueClass = outputValue.getClass();
-          outputValueType = FunctionUtils.getArgumentType(outputValueClass);
-          Preconditions.checkState(outputValueType != null, "Unsupported output value class: %s", outputValueClass);
+        try {
+          Object outputValue = functionEvaluator.evaluate(inputValues);
+          outputValues[i] = outputValue;
+          if (outputValueType == null) {
+            Class<?> outputValueClass = outputValue.getClass();
+            outputValueType = FunctionUtils.getArgumentType(outputValueClass);
+            Preconditions.checkState(outputValueType != null, "Unsupported output value class: %s", outputValueClass);
+          }
+        } catch (Exception e) {
+          if (!errorOnFailure && hasNull) {
+            LOGGER.warn("Caught exception while evaluating derived column: {} with function: {} and arguments: {}",
```

Comment:
This can easily flood the log (one per row when an invalid transform is provided)

Reply:
write a sampler for this.